plugins/pipelines/update.js

```diff
@@ -18,15 +18,18 @@ const ANNOTATION_USE_DEPLOY_KEY = 'screwdriver.cd/useDeployKey';
  * @return {Promise}
  */
 function getPermissionsForOldPipeline({ scmContexts, pipeline, user }) {
+    const isPipelineSCMContextObsolete = !scmContexts.includes(pipeline.scmContext);
+
     // this pipeline's scmContext has been removed, allow current admin to change it
-    if (!scmContexts.includes(pipeline.scmContext)) {
-        const permission = { admin: false };
+    // also allow pipeline admins from other scmContexts to change it
+    if (isPipelineSCMContextObsolete || user.scmContext !== pipeline.scmContext) {
+        let isAdmin = pipeline.adminUserIds.includes(user.id) || !!pipeline.admins[user.username];
 
-        if (pipeline.admins[user.username]) {
-            permission.admin = true;
+        if (!isAdmin && isPipelineSCMContextObsolete && pipeline.admins[user.username]) {
+            isAdmin = true;
         }
 
-        return Promise.resolve(permission);
+        return Promise.resolve({ admin: isAdmin });
```

Comment:
`const isDifferentSCMContext = user.scmContext !== pipeline.scmContext;

    let isAdmin = false;

    if (isPipelineSCMContextObsolete || isDifferentSCMContext) {
        const isUserIdInAdminList = pipeline.adminUserIds.includes(user.id);
        const isUsernameInAdminsObject = !!pipeline.admins[user.username];
        const isObsoleteAdmin = isPipelineSCMContextObsolete && pipeline.admins[user.username];

        isAdmin = isUserIdInAdminList || isUsernameInAdminsObject || isObsoleteAdmin;
    }

    return Promise.resolve({ admin: isAdmin });`

Reply:
the code is a bit difficult to read in terms of what the logic is, hence suggesting detailed variable names